### PR TITLE
Use workload-identity to deploy prow

### PR DESCRIPTION
**Pull request by fejta**
ref https://github.com/GoogleCloudPlatform/oss-test-infra/issues/202

### CODE REVIEW

File: prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml [hunk 1]
```diff
@@ -64,6 +64,7 @@ postsubmits:
     labels:
       preset-prow-deployer-service-account: "true"
```
Comment (cjwagner) [inline prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml]: Remove this too? (Is this even an extant preset?)

File: prow/serviceaccounts/GoogleCloudPlatform_oss-test-infra_serviceaccounts.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
```
Comment (cjwagner) [inline prow/serviceaccounts/GoogleCloudPlatform_oss-test-infra_serviceaccounts.yaml]: Please make this file deploy automatically.

Comment (fejta) [reply prow/serviceaccounts/GoogleCloudPlatform_oss-test-infra_serviceaccounts.yaml]: Done

File: prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml (hunk 1 above)
Comment (fejta) [reply prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml]: Done

**Pull request merged**